pkg/operator/configobservation/apiserver/observe_send_retry_after_while_not_ready_once.go

```diff
@@ -0,0 +1,76 @@
+package apiserver
+
+import (
+	"fmt"
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const sendRetryAfterWhileNotReadyOnceKey = "send-retry-after-while-not-ready-once"
+
+// ObserveSendRetryAfterWhileNotReadyOnce ensures that send-retry-after-while-not-ready-once is set for SNO clusters.
+func ObserveSendRetryAfterWhileNotReadyOnce(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains apiServerArguments field.
+		ret = configobserver.Pruned(ret, []string{"apiServerArguments"})
+	}()
+
+	// read the observed value
+	listers := genericListers.(configobservation.Listers)
+	infra, err := listers.InfrastructureLister().Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// we got an error so without the infrastructure object we are not able to determine the type of platform we are running on
+		return existingConfig, append(errs, err)
+	}
+
+	// read the current value
+	expectedApiServerArguments := make(map[string]interface{}, 0)
+	currentApiServerArguments, _, err := unstructured.NestedMap(existingConfig, "apiServerArguments")
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to extract apiserver arguments from the existing config: %v", err))
+		// without a full set of apiserver arguments we can't proceed
+		return existingConfig, append(errs, err)
+	}
+	// shallow copy current apiserver arguments to expected
```

Comment:
nit: so on any error, we are not continuing, can we 
```
	if err != nil {
		  if !apierrors.IsNotFound(err) {
                        return existingConfig, append(errs, err)
                  }
                 errs = append(errs, fmt.Errorf("unable to extract apiserver arguments from the existing config: %v", err))
                return existingConfig, errs
	}
```

why is NotFound special, can't we exit when there is any error?